examples/autograd/README.md

```diff
@@ -0,0 +1,20 @@
+# Basic data flow through autograd
+
+This example contains a very simple multi layer model meant to understand how autograd is used in Hasktorch to calculate gradients. The weights are taken to be equal to 1, and no activation has been applied to allow the user to conveniently verify the gradients for themselves. In the absence of the activation being applied and when the weights are equal to 1, the gradients for the respective weights will be equal to the value of input feature they are connected to. 
```

Comment:
Can we please add an activation function? It doesn't have to be super complex, but differentiating through a linear computation is not the most impressive thing you can do. If you think that might be helpful, you could include the hand-derivation of the gradient to explain why does the user see some output values.

Reply:
Yeah, that would make the example more effective. I will add an activation function, along with a derivation of the gradients.